shared/actions/chat2/team-building.js

```diff
@@ -0,0 +1,123 @@
+// @flow
+import logger from '../../logger'
+import * as TeamBuildingTypes from '../../constants/types/team-building'
+import * as TeamBuildingGen from '../team-building-gen'
+import * as Chat2Gen from '../chat2-gen'
+import * as RouteTreeGen from '../route-tree-gen'
+import * as Saga from '../../util/saga'
+import * as RPCTypes from '../../constants/types/rpc-gen'
+import {trim} from 'lodash-es'
+import {type TypedState} from '../../constants/reducer'
+
+type RawSearchResult = {
+  score: number,
+  keybase: ?{
+    username: string,
+    uid: string,
+    picture_url: ?string,
+    full_name: ?string,
+    is_followee: boolean,
+  },
+  service: ?{
+    service_name: TeamBuildingTypes.ServiceIdWithContact,
+    username: string,
+    picture_url: ?string,
+    bio: ?string,
+    location: ?string,
+    full_name: ?string,
+  },
+  services_summary: {
+    [key: TeamBuildingTypes.ServiceIdWithContact]: {
+      service_name: TeamBuildingTypes.ServiceIdWithContact,
+      username: string,
+    },
+  },
+}
+
+const closeTeamBuilding = () => Saga.put(RouteTreeGen.createNavigateUp())
+
+const search = (_, {payload: {query, service, limit = 10}}: TeamBuildingGen.SearchPayload) => {
```

Comment:
the main motivation for this form of actionToAction and actionToPromise is that usually the data we want is likely in the store. so if we go down the route of having one result, instead of the map, you could have the reducer pull the query and store it in the store, then you'd get it from tehre and not the raw action